spec/menu_spec.rb

```diff
@@ -0,0 +1,45 @@
+require 'menu'
+
+describe Menu do
+  let(:dish_price) { rand(1..5) }
+  let(:quantity) { rand(1..10) }
+  let(:dish_total) { dish_price * quantity }
+  let(:dish) { double("dish", :read_name => "dish", :read_price => dish_price, :amount => quantity, :dish_total => dish_total) }
+  let(:dish2) { double("dish2", :read_name => "dish2", :read_price => dish_price, :amount => quantity, :dish_total => dish_total) }
```

Comment:
This is fantastic! Such a good use of let and doubles to test with flexibility rather than hardcoded dishes and prices.  I've learnt something here already! Nice one.